src/request/providers/node.ts

```diff
@@ -88,7 +88,7 @@ interface HttpsOptions extends Options {
 interface RequestData {
 	task: Task<http.IncomingMessage>;
 	buffer: any[];
-	data: string;
+	data: Buffer | string;
```

Comment:
Since data should only ever be a buffer (the only thing that's assigned to it is `dataAsBuffer`), should its type here be Buffer? The initializer would then need to change to use something like `Buffer.alloc(0)`.